pkgs/tools/networking/imapsync/default.nix

```diff
@@ -1,10 +1,13 @@
-{lib, stdenv, makeWrapper, fetchurl, perl, openssl, perlPackages }:
+{lib, stdenv, makeWrapper, fetchFromGitHub, perl, openssl, perlPackages, procps }:
 
 stdenv.mkDerivation rec {
-  name = "imapsync-1.727";
-  src = fetchurl {
-    url = "https://releases.pagure.org/imapsync/${name}.tgz";
-    sha256 = "1axacjw2wyaphczfw3kfmi5cl83fyr8nb207nks40fxkbs8q5dlr";
+  pname = "imapsync";
+  version = "1.977";
+  src = fetchFromGitHub {
+    owner = pname;
+    repo = pname;
+    rev = version;
+    sha256 = "1a9dpnzmi2lhqwj3skvxl62kc911my0x64x2djqn89cj52c5lhaa";
   };
 
   patchPhase = ''
```

Comment:
```suggestion
  patchPhase = ''
    sed -i -e s@/usr@$out@ Makefile
    sed -i '/PAR::Packer/d' INSTALL.d/prerequisites_imapsync
    sed -i '/App::cpanminus/d' INSTALL.d/prerequisites_imapsync
    sed -i '/^test_cpanm$/d' INSTALL.d/prerequisites_imapsync
  '';
```

Reply:
`PAR::Packer` and `App::cpanminus` are not hard dependencies, `imapsync` works fine without them.

Reply:
I was totally blind in assuming those would be hard deps - this is fixed now using patches as:
- `substituteInPlace` should usually be used instead of `sed`
- `substituteInPlace` doesn't a non-hackish/stable way of dropping the `test_cpanm` execution (although this actually wouldn't have mattered, as its returncode isn't checked anyways)
- dropping `patchPhase` also made sense due to 5b3d2ad8a08f3d96a137d35685111a306f111b78
- another minor patch for `Makefile` to create the required `W` directory was needed as well